src/modules/tas_studio/mod.rs

```diff
@@ -385,6 +387,30 @@ fn convert_hltas_from_bxt_tas_new(marker: MainThreadMarker, mut path: String) {
     )
 }
 
+fn norefresh_until_stop_frame_frame_idx(marker: MainThreadMarker, editor: &Editor) -> usize {
+    let norefresh_last_frames = unsafe { bxt::BXT_TAS_NOREFRESH_UNTIL_LAST_FRAMES.get(marker)() };
+    (editor.stop_frame() as usize).saturating_sub(norefresh_last_frames as usize)
+}
+
+fn set_effective_norefresh_until_stop_frame(marker: MainThreadMarker, editor: &Editor) {
+    let stop_frame = norefresh_until_stop_frame_frame_idx(marker, editor);
+
+    if stop_frame == 0 {
```

Comment:
Should this logic be in norefresh_until_stop_frame_frame_idx() instead?

Reply:
It is very possible for those two functions to be just one because the set function does not return anything. But I still have both of them into different thing.

As for the stop frame logic thing, it is a rare case when stop frame is 0, which is usually full tas playback or starting to do the tas. It could just be a simple return but for some reasons it stop frame 0 and until last frames xxx does not work though I reset the override value after playback is done in BunnymodXT (so stop frame 0 will automatically pickup the until last frames value). That is why the extra set call i there. 

Maybe I don't understand what you are saying here.

Reply:
What I am saying is, maybe the `if stop_frame == 0 { subtract from total frames } else { subtract from stop frame }` logic should be inside `norefresh_until_stop_frame_frame_idx()`?